android/app/src/main/java/org/openbot/env/Vehicle.java

```diff
@@ -156,34 +167,42 @@ public void connectUsb() {
   }
 
   public void disconnectUsb() {
-    Objects.requireNonNull(usbConnection)
-        .send(
-            String.format(
-                Locale.US,
-                "c%d,%d\n",
-                (int) (getControl().getLeft()),
-                (int) (getControl().getRight())));
-
-    Objects.requireNonNull(usbConnection).stopUsbConnection();
-    usbConnection = null;
-    usbConnected = false;
+    if (usbConnection != null) {
+      usbConnection.send(
+          String.format(Locale.US, "c%d,%d\n", (int) (getLeftSpeed()), (int) (getRightSpeed())));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.